### PR TITLE
Prevent division by zero in HWBA to HSVA conversions

### PR DESCRIPTION
**Pull request by ickshonpe**
# Problem

Division by zero in `crates/bevy_color/src/hsva.rs` when `blackness` is `1`:

```rust
impl From<Hwba> for Hsva {
    fn from(
        Hwba {
            hue,
            whiteness,
            blackness,
            alpha,
        }: Hwba,
    ) -> Self {
        // Based on https://en.wikipedia.org/wiki/HWB_color_model#Conversion
        let value = 1. - blackness;
        let saturation = 1. - (whiteness / value);

        Hsva::new(hue, saturation, value, alpha)
    }
}
```

## Solution
With `Hsva` colors if the `value` component is set to `0.`  the output will be pure black regardless of the values of the `hue` or `saturation` components.

So if `value` is `0`, we don't need to calculate a `saturation` value and can just set it to `0`:

```rust
impl From<Hwba> for Hsva {
    fn from(
        Hwba {
            hue,
            whiteness,
            blackness,
            alpha,
        }: Hwba,
    ) -> Self {
        // Based on https://en.wikipedia.org/wiki/HWB_color_model#Conversion
        let value = 1. - blackness;
        let saturation = if value != 0. {
            1. - (whiteness / value)
        } else {
            0.
        };

        Hsva::new(hue, saturation, value, alpha)
    }
}

```

### CODE REVIEW

File: crates/bevy_color/src/hsva.rs [hunk 1]
```diff
@@ -157,7 +157,11 @@ impl From<Hwba> for Hsva {
     ) -> Self {
         // Based on https://en.wikipedia.org/wiki/HWB_color_model#Conversion
         let value = 1. - blackness;
-        let saturation = 1. - (whiteness / value);
+        let saturation = if value != 0. {
+            1. - (whiteness / value)
+        } else {
+            0.
+        };
```
Comment (MrGVSV) [inline crates/bevy_color/src/hsva.rs:164]: Could you used `checked_div` for this?
```suggestion
        let saturation = 1. - whiteness.checked_div(value).unwrap_or(1.);
```

Comment (ickshonpe) [reply crates/bevy_color/src/hsva.rs:164]: yep same behaviour in both, definite improvment.

Comment (ickshonpe) [reply crates/bevy_color/src/hsva.rs]: Oh wait no such method?

Comment (MrGVSV) [reply crates/bevy_color/src/hsva.rs:164]: Oops my bad. I could've sworn it existed for floats too 😅

**Review by MrGVSV — APPROVED** [review]

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**